frontend/src/app/main/data/workspace/selection.cljs

```diff
@@ -359,88 +361,113 @@
-  ([changes page component-root delta libraries library-data it]
-   (let [component-id (:component-id component-root)
-         file-id (:component-file component-root)
-         main-component    (ctf/get-component libraries file-id component-id)
-         moved-component   (gsh/move component-root delta)
-         pos               (gpt/point (:x moved-component) (:y moved-component))
-
-         instantiate-component
-         #(dwlh/generate-instantiate-component changes
-                                               file-id
-                                               (:component-id component-root)
-                                               pos
-                                               page
-                                               libraries
-                                               (:id component-root))
-
-         restore-component
-         #(let [restore (dwlh/prepare-restore-component library-data (:component-id component-root) it page delta (:id component-root) changes)]
-            [(:shape restore) (:changes restore)])
-
-         [_shape changes]
-         (if (nil? main-component)
-           (restore-component)
-           (instantiate-component))]
-     changes)))
+  [changes page component-root delta libraries library-data it]
+  (let [component-id    (:component-id component-root)
+        file-id         (:component-file component-root)
+        main-component  (ctf/get-component libraries file-id component-id)
+        moved-component (gsh/move component-root delta)
+        pos             (gpt/point (:x moved-component) (:y moved-component))
+
+        instantiate-component
+        #(dwlh/generate-instantiate-component changes
+                                              file-id
+                                              (:component-id component-root)
+                                              pos
+                                              page
+                                              libraries
+                                              (:id component-root))
+
+        restore-component
+        #(let [restore (dwlh/prepare-restore-component library-data (:component-id component-root) it page delta (:id component-root) changes)]
+           [(:shape restore) (:changes restore)])
+
+        [_shape changes]
+        (if (nil? main-component)
+          (restore-component)
+          (instantiate-component))]
+    changes))
+
+;; Return true if the object is a component that exists on the file or its libraries (even a deleted one)
+(defn- is-known-component?
+  [obj libraries library-data]
+  (let [main-instance?    (ctk/main-instance? obj)
+        component-id      (:component-id obj)
+        file-id           (:component-file obj)
+        main-component    (ctf/get-component libraries file-id component-id)
+        deleted-component (ctkl/get-deleted-component library-data component-id)]
+    (and main-instance?
+         (or main-component deleted-component))))
```

Comment:
This probably should be a helper in `common.types.file`

Reply:
Fixed